src/BloomExe/web/ServerBase.cs

```diff
@@ -106,6 +106,7 @@ private void TryStart()
 			for (var i = 0; i < _workers.Length; i++)
 			{
 				_workers[i] = new Thread(RequestProcessorLoop);
+				_workers[i].Priority = ThreadPriority.BelowNormal;
```

Comment:
I suppose you've tried with and without this...but these threads are not just used for making thumbnails in the background, right? They are also used to retrieve all the data for the current page. Does that cause any problems? would it be feasible to determine (e.g., from the URL) that a particular request is for thumbnail updating, and lower the priority of the thread assigned to handle it for the duration of that call? Or pick a lower-priority thread from a different collection?